internal/resolution/entitysources/catalogdsource.go

```diff
@@ -0,0 +1,142 @@
+
+	catalogd "github.com/operator-framework/catalogd/pkg/apis/core/v1beta1"
+)
+
+// catalogdEntitySource is a source for(/collection of) deppy defined input.Entity, built from content
+// made accessible on-cluster by https://github.com/operator-framework/catalogd.
+// It is an implementation of deppy defined input.EntitySource
+type catalogdEntitySource map[deppy.Identifier]input.Entity
+
+func NewCatalogdEntitySource(ctx context.Context, client client.Client) (catalogdEntitySource, error) {
+	entities := catalogdEntitySource{}
+	bundleMetadatas, packageMetdatas, err := onClusterMetadata(ctx, client)
+	if err != nil {
+		return nil, err
+	}
+	for _, bundle := range bundleMetadatas.Items {
+		props := map[string]string{}
+
+		for _, prop := range bundle.Spec.Properties {
+			switch prop.Type {
+			case property.TypePackage:
+				type pkg struct {
+					Name    string `json:"packageName"`
+					Version string `json:"version"`
+				}
+				tmp := pkg{}
+				if err := json.Unmarshal(prop.Value, &tmp); err != nil {
+					return nil, err
+				}
+				pkgValue, err := jsonMarshal(tmp)
+				if err != nil {
+					return nil, err
+				}
+				props[property.TypePackage] = string(pkgValue)
+			}
+		}
+
+		imgValue, err := jsonMarshal(bundle.Spec.Image)
+		if err != nil {
+			return nil, err
+		}
+		props["olm.bundle.path"] = string(imgValue)
+		bundlePkg := packageMetdatas[bundle.Spec.Package]
+		channelName := ""
+		for _, ch := range bundlePkg.Spec.Channels {
+			for _, b := range ch.Entries {
+				if b.Name == bundle.Name {
+					channelName = ch.Name
+				}
+			}
+		}
+		channelValue, _ := jsonMarshal(property.Channel{ChannelName: channelName, Priority: 0})
+		props[property.TypeChannel] = string(channelValue)
+		entity := input.Entity{
+			ID:         deppy.IdentifierFromString(fmt.Sprintf("%s%s%s", bundle.Name, bundle.Spec.Package, channelName)),
+			Properties: props,
+		}
+		entities[entity.ID] = entity
+	}
+	return entities, nil
+}
+
+func (es catalogdEntitySource) Get(ctx context.Context, id deppy.Identifier) *input.Entity {
+	if entity, ok := es[id]; ok {
+		return &entity
+	}
+	return nil
+}
+
+func (es catalogdEntitySource) Filter(ctx context.Context, filter input.Predicate) (input.EntityList, error) {
+	resultSet := input.EntityList{}
+	for _, entity := range es {
+		if filter(&entity) {
+			resultSet = append(resultSet, entity)
+		}
+	}
+	return resultSet, nil
+}
+
+func (es catalogdEntitySource) GroupBy(ctx context.Context, fn input.GroupByFunction) (input.EntityListMap, error) {
+	resultSet := input.EntityListMap{}
+	for _, entity := range es {
+		keys := fn(&entity)
+		for _, key := range keys {
+			resultSet[key] = append(resultSet[key], entity)
+		}
+	}
+	return resultSet, nil
+}
+
+func (es catalogdEntitySource) Iterate(ctx context.Context, fn input.IteratorFunction) error {
+	for _, entity := range es {
+		if err := fn(&entity); err != nil {
+			return err
+		}
+	}
+	return nil
+}
```

Comment:
> The biggest change is that it makes the list call and builds entities every time these calls are made. A future optimization would be to implement caching

So what is the objective we're trying to achieve with this, today? This definitely is an interesting idea. My brain just molded into the format of the two entitySources I observed before this (removed as a part of this PR, and replaced with the catalogd entitysource), but I'm not sure I understand the value of doing this in achieve the goal "first pass over of the integration with catalogd"

Reply:
> So what is the objective we're trying to achieve with this, today?

The main objective is to not have to instantiate a new entity source every single time you want to resolve.

>My brain just molded into the format of the two entitySources I observed before this (removed as a part of this PR, and replaced with the catalogd entitysource)

I'm not sure I'm following here - from what I could tell the way that the two entity sources before this one were both designed with the idea in mind that you create it once not create it every time you need to resolve something. In my mind this implementation seems to change the pattern introduced by the other entity sources. The hardcoded entity source was only "static" in the sense that it hardcoded its responses but appeared to be "dynamic" to the controller since the idea was you instantiate it once and use it everywhere. The same goes for the OLMv0 integration - it used a controller to keep track of `CatalogSource` CRs and update its cache accordingly.

Maybe the verbiage of "static" and "dynamic" is misleading here but what I am meaning is that *I* feel that we should be following the pattern set by the previous entity sources of using one instantiation everywhere instead of instantiating every single time we want to use the resolver.

> but I'm not sure I understand the value of doing this in achieving the goal "first pass over of the integration with catalogd"

I don't think there is much difference in value but from what I can tell, just that this PR appears to change the pattern established by the previous two entity sources

Reply:
Another note on this - based on [this comment](https://github.com/operator-framework/operator-controller/pull/141#issuecomment-1516407385) I don't think we are following the "on the fly" approach proposed since we are loading everything at instantiation and instantiating it every time we need to resolve. 

*Maybe* it falls into the optimization territory but I think taking the approach of instantiate once and use a client to fetch when certain functions are called is more aligned with what was proposed by the comment I linked